Server/mods/deathmatch/logic/CGame.cpp

```diff
@@ -4699,3 +4705,31 @@ void CGame::HandleCrashDumpEncryption()
     }
 #endif
 }
+
+void CGame::RegisterClientTriggeredEventUsage(CPlayer* pPlayer)
+{
+    if (!pPlayer)
+        return;
+
+    auto iter = m_mapClientTriggeredEvents.find(pPlayer);
+
+    if (iter == m_mapClientTriggeredEvents.end())
+        iter = m_mapClientTriggeredEvents.insert({pPlayer, 0}).first;
+
+    iter->second++;
```

Comment:
You don't decrease this value with time. It's works not how you describe/expecting. It cause _random_ triggering after any random spam in time.

Reply:
The value gets wiped every second in `CGame::ProcessClientTriggeredEventSpam`. There is a possibility the player will get kicked if they go over the threshold before the interval is over, but they would be getting kicked anyway...

Reply:
My bad. I see this now